dali/operators/decoder/audio/audio_decoder_impl.cc

```diff
@@ -17,39 +17,57 @@
 
 namespace dali {
 
-template <typename T>
-span<char> as_raw_span(T *buffer, ptrdiff_t length) {
-  return make_span(reinterpret_cast<char*>(buffer), length*sizeof(T));
+std::pair<int64_t, int64_t> ProcessOffsetAndLength(const AudioMetadata &meta,
+                                                   double offset_sec, double length_sec) {
+  int64_t offset = 0;
+  int64_t length = meta.length;
+  if (offset_sec >= 0.0f) {
```

Comment:
```suggestion
  if (offset_sec >= 0.0) {
```
Comparing double to double